Update README to use csctl

dudleycarr opened the pull request.

lightbend-cla-validator commented:
Hi @dudleycarr,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[https://www.lightbend.com/contribute/cla](https://www.lightbend.com/contribute/cla)

lightbend-cla-validator commented:
Hi @dudleycarr,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[https://www.lightbend.com/contribute/cla](https://www.lightbend.com/contribute/cla)

rstento commented on README.md:
```diff
@@ -3,230 +3,161 @@
+## Deploying
 
-## Sample application layout
+The following steps uses `csctl` to deploy both the [shopping-cart frontend
```
A few typos corrected here, corrections in italics:

The following steps *use* `csctl` to deploy both the [shopping-cart frontend
service](https://github.com/cloudstateio/samples-ui-shoppingcart) *and* the
CloudState shopping-cart service defined in this repo. We'll be using the
pre-built Docker images for the frontend and the shopping-cart, but feel free to
build and use your own Docker images where appropriate.

rstento commented on README.md:
````diff
@@ -3,230 +3,161 @@
+```
 
-### Frontend Service
+Wait until you get an email about your project being created and approved!
````
How about:

Wait until you receive an email approving your project!

rstento commented on README.md:
````diff
@@ -3,230 +3,161 @@
+```
 
-### Stateful Store
+Proceed when `STATUS` is `ready`. This will take approximately 5 minutes.
````
I always hesitate to put a time since I suspect it might vary. How about, Proceed when the project `STATUS` is `ready`, this can take some time.

vkorenev commented on README.md:
````diff
@@ -3,230 +3,161 @@
 
-```bash
-$ cd deploy
-$ kubectl apply -f . -n <project-name>
-# verify stateful store
-$ kubectl get -n <project-name> statefulstore
````
As far as I know, we would like to provide 2 sets of instructions: one using `csctl` for Lightbend Cloudstate hosting and one using `kubectl` for self-hosted open-source Cloudstate. So these instructions are still needed.

dudleycarr replied:
OK, I can definitely add that back in.

vkorenev replied:
@rstento how would we like to organize those 2 sets of instructions? I like tabs like here https://docs.lbcs.dev/administration/log.html We can use them when we eventually convert the instructions to AsciiDoc.
@michaelpnash has mentioned that we may also want to use tables.

dudleycarr replied:
One option would be to put the self-hosted instructions in the deploy directory and link to that from the top-level README.

rstento replied:
@vkorenev & @dudleycarr, I like the idea of putting the self-hosted instructions in a separate file that is linked to from the readme for now. Once we get the asciidoc set up, we can play around with the best presentation, perhaps tabs, perhaps a table. Sound OK?

vkorenev replied:
@dudleycarr @rstento, yes, I think it is good to place self-hosted instructions in a separate file. Actually, there are 3 logical sections which we could put in separate files:
- local development and building image(s)
- deploying on Lightbend Cloudsate
- deploying on generic k8s cluster

rstento commented:
When we have asciidoc set up, we can include the set up local dev env and
deploy instructions from the cloudstate oss doc.

On Fri, Jul 10, 2020 at 2:24 PM Vladimir Korenev <notifications@github.com>
wrote:

> *@vkorenev* commented on this pull request.
> ------------------------------
>
> In README.md
> <https://github.com/cloudstateio/samples-js-shoppingcart/pull/9#discussion_r453036805>
> :
>
> > -$ kubectl apply -f . -n <project-name>
> -# verify stateful store
> -$ kubectl get -n <project-name> statefulstore
>
> @dudleycarr <https://github.com/dudleycarr> @rstento
> <https://github.com/rstento>, yes, I think it is good to place
> self-hosted instructions in a separate file. Actually, there are 3 logical
> sections which we could put in separate files:
>
>    - local development and building image(s)
>    - deploying on Lightbend Cloudsate
>    - deploying on generic k8s cluster
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cloudstateio/samples-js-shoppingcart/pull/9#discussion_r453036805>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFOUG62DYUGK7B4ZY63Y6CLR25THBANCNFSM4OWVZ5LA>
> .
>
-- 
Ruth Stento
User Experience Lead
Lightbend

512-696-1448
<http://lightbend.com>

dudleycarr commented:
I've added the previous k8s instructions back in and they now live in the deploy directory README. I removed some items from those instructions that either reference Lightbend CloudState which is inaccurate for self-hosting.

The k8s + CloudState instructions currently lack the last step for how to access the app after having the routes deployed.  I suggest we create an issue to address that bit of missing instruction and get this merged up if we're otherwise happy with the structure.

vkorenev commented:
Actually, the route creation step does not work on a generic K8s cluster. It was working on Lightbend Cloudstate only.
Anyway, I think it is good to merge this PR now and address that later.
